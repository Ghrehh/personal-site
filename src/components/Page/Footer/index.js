// @flow
import * as React from 'react';

import SectionHeading from 'components/text/SectionHeading';
import Button from 'components/Button';

import styles from './styles.module.css';

const Footer = () => (
  <React.Fragment>
    <div className={styles.slantedBackground} />

    <footer className={styles.footer}>
      <div className={styles.footerInner}>
        <div className={styles.textContainer}>
          <SectionHeading className={styles.heading}>CONTACT</SectionHeading>
          <h2 className={styles.subHeading}>Feel free to get in touch.</h2>
        </div>

        <div>
          <Button className={styles.twitterButton}>Twitter</Button>
          <Button className={styles.emailButton}>Email</Button>
          <Button className={styles.linkedInButton}>LinkedIn</Button>
        </div>
      </div>
    </footer>
  </React.Fragment>
);

export default Footer;
