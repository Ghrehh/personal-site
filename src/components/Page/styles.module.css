.page {
  padding-top: 50px;
}

.main {
  min-height: 100vh;
}
