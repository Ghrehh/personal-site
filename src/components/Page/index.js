// @flow
import * as React from 'react';

import Footer from './Footer';
import styles from './styles.module.css';

type Props = {
  children: React.Node,
  style?: {}
};

const Page = (props: Props) => (
  <div style={props.style} className={styles.page}>
    <main className={styles.main}>{props.children}</main>
    <Footer />
  </div>
);

export default Page;
