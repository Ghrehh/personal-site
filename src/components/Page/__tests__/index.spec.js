import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import Page from '../';

describe('Page', () => {
  let testContext = {};

  beforeEach(() => {
    testContext = {};
    testContext.renderer = new ShallowRenderer();
    testContext.renderer.render(
      <Page style={{ backgroundColor: 'blue' }}>
        <p>foo</p>
      </Page>
    );
    testContext.result = testContext.renderer.getRenderOutput();
  });

  it('renders the component correctly', () => {
    expect(testContext.result).toMatchSnapshot();
  });
});
