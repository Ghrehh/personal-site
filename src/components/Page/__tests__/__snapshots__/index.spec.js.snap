// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page renders the component correctly 1`] = `
<div
  className="page"
  style={
    Object {
      "backgroundColor": "blue",
    }
  }
>
  <main
    className="main"
  >
    <p>
      foo
    </p>
  </main>
  <Footer />
</div>
`;
